src/xtgeo/grid3d/grid_property.py

```diff
@@ -71,39 +90,98 @@ def _data_reader_factory(fformat):
     elif fformat == "bgrdecl":
         return import_bgrdecl_prop
 
-    elif fformat in ["xtg"]:
+    elif fformat == "xtg":
         return import_xtgcpprop
     else:
         raise ValueError(f"Invalid grid property file format {fformat}")
 
 
-def gridproperty_from_file(*args, **kwargs):
-    """Make a GridProperty instance directly from file import.
+def gridproperty_from_file(
+    pfile: Union[str, pathlib.Path, io.BytesIO, io.StringIO],
+    fformat: Optional[str] = None,
+    **kwargs: Any,
```

Comment:
Need to double check with Jan about a complete list of these keyword args. As we are deprecating other means of initializing from a file the available arguments should be fully documented here as parameters.

Reply:
You can use `Dict[str, Any]`, unsure if it adds any extra information.